demisto_sdk/commands/download/downloader.py

```diff
@@ -95,647 +87,851 @@
         """
-        Fetches the custom content from Demisto into a temporary dir.
-        :return: True if fetched successfully, False otherwise
+        Download custom content bundle using server's API,
+        and create a StringIO object containing file data for each file within it.
+
+        Returns:
+            dict[str, StringIO]: A dictionary mapping custom content's file names to file objects.
         """
+        # Set to 'verify' to None so that 'demisto_client' will use the environment variable 'DEMISTO_VERIFY_SSL'.
+        verify = not self.insecure if self.insecure else None
+        logger.info("Fetching custom content bundle from server...")
+
         try:
-            verify = (
-                (not self.insecure) if self.insecure else None
-            )  # set to None so demisto_client will use env var DEMISTO_VERIFY_SSL
             self.client = demisto_client.configure(verify_ssl=verify)
-            api_response: tuple = demisto_client.generic_request_func(
-                self.client, "/content/bundle", "GET"
-            )
-            body: bytes = ast.literal_eval(api_response[0])
-            io_bytes = io.BytesIO(body)
-
-            # Demisto's custom content file is of type tar.gz
-            tar = tarfile.open(fileobj=io_bytes, mode="r")
-
-            (
-                content_items_file_names_tuple,
-                scripts_id_to_name,
-            ) = self.find_uuids_in_content_item(tar)
-            self.write_custom_content(
-                content_items_file_names_tuple, scripts_id_to_name
-            )
-            return True
+            api_response: HTTPResponse = demisto_client.generic_request_func(
+                self.client,
+                "/content/bundle",
+                "GET",
+                _preload_content=False,
+            )[0]
 
-        except ApiException as e:
-            self.handle_api_exception(e)
-            return False
-        except MaxRetryError as e:
-            self.handle_max_retry_error(e)
-            return False
         except Exception as e:
-            logger.info(f"Exception raised when fetching custom content:\n{e}")
-            return False
+            if isinstance(e, ApiException) and e.status == 401:
+                logger.error(
+                    f"Server authentication error: {e}\n"
+                    "Please verify that the required environment variables ('DEMISTO_API_KEY', or "
+                    "'DEMISTO_USERNAME' and 'DEMISTO_PASSWORD') are properly configured."
+                )
+
+            elif isinstance(e, MaxRetryError):
+                logger.error(
+                    f"Failed connecting to server: {e}.\n"
+                    "Please verify that the environment variable 'DEMISTO_BASE_URL' is properly configured, "
+                    "and that the server is accessible.\n"
+                    "If the server is using a self-signed certificate, try using the '--insecure' flag."
+                )
```

Comment:
Since there's some shared code between the different exceptions (`raise HandledError from e`), and I didn't want to repeat it for every `except` block (and then if someone wanted to change it, he would have to change the same line in 3 except blocks).